director/builder.py

```diff
@@ -11,82 +13,85 @@
+        options["queue"] = params.get("queue", self.default_queue)
+        kwargs.update(workflow_id=self.workflow_id, payload=self.workflow.payload)
 
-    def new_task(self, task_name, single=True):
-        task_id = uuid()
+        # Create a Celery task specifying its UUID
+        task = cel.tasks.get(name)
+        signature = task.subtask(task_id=task_id, kwargs=kwargs, **options)
 
-        # We create the Celery task specifying its UID
-        signature = cel.tasks.get(task_name).subtask(
-            kwargs={"workflow_id": self.workflow_id, "payload": self.workflow.payload},
-            queue=self.queue,
-            task_id=task_id,
-        )
-
-        # Director task has the same UID
+        # Create a Director task with the same UUID
         task = Task(
             id=task_id,
-            key=task_name,
+            key=name,
+            queue=options["queue"],
             previous=self.previous,
-            workflow_id=self.workflow.id,
+            workflow_id=self.workflow_id,
             status=StatusType.pending,
-        )
-        task.save()
+        ).save()
 
         if single:
             self.previous = [signature.id]
 
         return signature
 
-    def parse(self, tasks):
+    def new_group(self, item):
+        params = {k: v for k, v in item.items() if k not in ['type', 'name', 'tasks']}
+        group_tasks = [self.new_task(**t, single=False) for t in item['tasks']]
+        self.previous = [s.id for s in group_tasks]
+        return celery.group(*group_tasks, task_id=str(uuid4()), **params)
+
+    def parse_simple_item(self, item):
+        to_obj = lambda x: {"name": x, "type": "task"}
+        if type(item) is str:
+            return to_obj(item)
+        if type(item) is dict:
+            name = next(iter(item))
+            item = {"name": name, **item[name]}
+            if item.get("tasks"):
+                item["tasks"] = [to_obj(x) for x in item["tasks"]]
+            return item
+
+    def parse(self):
         canvas = []
-
-        for task in tasks:
-            if type(task) is str:
-                signature = self.new_task(task)
+        for item in self.blueprint["tasks"]:
+            item = deepcopy(item)
+            if not self.complex:
+                item = self.parse_simple_item(item)
+            if item["type"] == "task":
+                signature = self.new_task(**item)
                 canvas.append(signature)
-            elif type(task) is dict:
-                name = list(task)[0]
-                if "type" not in task[name] and task[name]["type"] != "group":
-                    raise WorkflowSyntaxError()
-
-                sub_canvas_tasks = [
-                    self.new_task(t, single=False) for t in task[name]["tasks"]
-                ]
-
-                sub_canvas = group(*sub_canvas_tasks, task_id=uuid())
+                continue
+            if item["type"] == "group":
+                sub_canvas = self.new_group(item)
```

Comment:
If all this is dealing with old (simple) vs new (complex) blueprints/schemas, and adding quite a bit of complexity in the code... why don't we just recreate the old blueprints/schemas in the new complex format?

Then we can remove all the if complex? And not have to refer to each entry as a v generic named "item", but instead a task_group? Or similar...?